template/cordova/lib/package.js

```diff
@@ -215,13 +215,20 @@ module.exports.deployToDesktop = function (package, deployTarget) {
 
     return utils.getAppStoreUtils().then(function(appStoreUtils) {
         return getPackageName(path.join(__dirname, '..', '..')).then(function(pkgname) {
+
+            var oldArch;
             // uninstalls previous application instance (if exists)
             console.log('Attempt to uninstall previous application version...');
             return spawn('powershell', ['-ExecutionPolicy', 'RemoteSigned', 'Import-Module "' + appStoreUtils + '"; Uninstall-App ' + pkgname])
             .then(function() {
                 console.log('Attempt to install application...');
+                oldArch = process.env.PROCESSOR_ARCHITECTURE;
+                if (package.arch === 'x64') {
+                    process.env.PROCESSOR_ARCHITECTURE = 'AMD64';
+                }
                 return spawn('powershell', ['-ExecutionPolicy', 'RemoteSigned', 'Import-Module "' + appStoreUtils + '"; Install-App', utils.quote(package.script)]);
```

Comment:
If we throw here the PROCESSOR_ARCHITECTURE will remain changed. While this currently has no side effects, we probably want to have a catch promise and set the PROCESSOR_ARCHITECTURE to the original value since it's a shared value. LGTM otherwise.